### DISCUSSION TITLE
[gatsby-plugin-mdx] Cannot read property 'forEach' of undefined

### DISCUSSION THREAD

**Discussion by AleJo2995** [gatsbyjs/gatsby#32935] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Hi. I´m facing an issue the time I add the `gatsby-plugin-mdx` plugin to the gatsby-config.js: 

ERROR 

There was an error compiling the html.js component for the development server.
See our docs page on debugging HTML builds for help https://gatsby.dev/debug-html TypeError: Cannot read property 'forEach' of undefined

  35 |   }
  36 |
> 37 | mdxPlugins.forEach(({ guards = {}, components }) => {
     |            ^
  38 |   Object.entries(components).forEach(([componentName, Component]) => {
  39 |     if (componentsAndGuards[componentName]) {
  40 |       componentsAndGuards.push({ guard: guards[componentName], Component })

  WebpackError: TypeError: Cannot read property 'forEach' of undefined
  
  - wrap-root-element.js:37 
    [gatsby-theme-carbon]/[gatsby-plugin-mdx]/wrap-root-element.js:37:12
  
  - gatsby-ssr.js:1 
    [gatsby-theme-carbon]/[gatsby-plugin-mdx]/gatsby-ssr.js:1:1
  

not finished Building development bundle - 16.043s

I´m using `gatsby-source-git` to source documentation from another github repo.

### Reproduction Link

n/a

### Steps to Reproduce

1.
2.
3.
...

### Expected Result

I expect the build to run successfully and get the plugin working

### Actual Result

The build doesn´t start at all 

### Environment

```shell
OS: MacOs Big Sur 11.4
Gatsby: V2
npm: 6.14.13
node: v14.16.0
```

**Comment by LekoArts** [comment]
The error points to `gatsby-theme-carbon` - so if you're using that you'll need to check how to correctly use that and/or open an issue there.
